### DISCUSSION TITLE
How to get the workflow run ID?

### DISCUSSION THREAD

**Discussion by rffontenelle** [community/community#26965] [Actions]
<p>I would like to send a Telegram message containing the URL of a specific workflow run when it failed. I found how to send Telegram message (via appleboy/telegram-action), found the repository info using  <strong>${{ github.repository }}</strong> , but I still haven’t found how to get that specific workflow run ID.  By ID, I mean the <strong>12345</strong> in <a href="https://github.com/foo/bar/actions/12345" rel="nofollow noopener">https://github.com/foo/bar/actions/12345</a>.</p>
<p>How can I get the workflow run ID?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p><a class="mention" href="/rffontenelle">@rffontenelle</a> </p>
<p>Do you mean the run id in <a href="https://github.com/foo/bar/actions/12345" rel="nofollow noopener">https://github.com/foo/bar/actions/ <strong>runs</strong> /12345</a>.? </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/71f97d87ca69298a77f0d6bdb5b3f3d74567b3dd.png" data-download-href="/uploads/short-url/gggu3u14WQGdkYOfEzII6Rv1WBT.png?dl=1" title="run id.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181100986-07d4623f-5ade-459d-b826-1c4f1c3c2388.png" alt="run id.png" data-base62-sha1="gggu3u14WQGdkYOfEzII6Rv1WBT" width="690" height="431" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/71f97d87ca69298a77f0d6bdb5b3f3d74567b3dd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">run id.png</span><span class="informations">815×510 23.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>There is a run_id property of <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener">github context</a>. You could use ${{ github.run_id }} to get the run ID.  </p>

**Comment by Novodes** [comment]
In case you need to extract it from the command line, you can use this script in bash:

```bash
RUNID=$(gh api repos/$OWNER/$REPO/commits/$COMMIT_SHA/check-runs | jq -r '.check_runs[] | select(.name == "NAME_OF_YOUR_GITHUB_ACTIONS_WORKFLOW") | .html_url | capture("/runs/(?<number>[0-9]+)/jobs") | .number' | sed 's/"//g' | head -n 1)
```

Full explanation on StackOverflow: https://stackoverflow.com/a/76465815/4441211

**Reply by Calinou** [reply]
I've tried that one-liner to get the run ID of an older commit within https://github.com/godotengine/godot, but I can't get it to work (the variable is empty).

This is what I tried:

```
RUNID=$(gh api repos/godotengine/godot/commits/94dbf69f5d6b7d2fd9561692df2e71557607fddc/check-runs | jq -r '.check_runs[] | select(.name == "🔗 GHA") | .html_url | capture("/runs/(?<number>[0-9]+)/jobs") | .number' | sed 's/"//g' | head -n 1)
```

`🔗 GHA` is the name of the main workflow file I'm interested in (it's defined in `.github/workflows/runner.yml`). I want to get a list of its artifacts so I can download them with [nightly.link](https://nightly.link/) and compare their binary sizes with a PR build (so I can't rely on a PAT, hence the use of nightly.link). There are premade actions out there to compare bundle sizes but here, I'm comparing sizes of compiled C++ binaries that take a long time to compile (so building them twice isn't an option). There's also an action to compare an artifact's size to a GitHub release's size, but the binaries I'm comparing against aren't uploaded to GitHub Releases, only artifacts.

Running this alone results in a lot of forks appearing in the list, which surprises me:

```
gh api repos/godotengine/godot/commits/94dbf69f5d6b7d2fd9561692df2e71557607fddc/check-runs
```

**Comment by jayeshkumarpatel2611** [comment]
How we can obtain run id if we dispatch workflow using api in powershell script?

**Reply by Novodes** [reply]
Doesn't this answer your question: https://github.com/orgs/community/discussions/26965#discussioncomment-6164539

**Reply by seanorama** [reply]
No. A GitHub workflow can be run directly with a dispatch event.

In that case, there is no commit to build on.

This is an incredibly large oversight, that a workflow system does not return anything about it's run. It's not a useful workflow system if you can't programmatically track the result.

As seen in the API docs, a worfklow dispatch only ever returns:
```
Status: 204
```

You expect any workflow/job/... based API to respond the ID of the thing it created.

https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event

**Reply by xzrderek** [reply]
totally agree, i would've expected a workflow dispatch to return `run_id` so we can correlate it back.

**Reply by seanorama** [reply]
It's basic asynchronous API development. You submit a request, you get back the id. Any solution that requires filters on other values is simply a workaround.

**Comment by infinisil** [comment]
A less hacky alternative to the above is to use the [API endpoint for listing workflow runs](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-workflow) and filtering by associated commit:

```bash
gh api /repos/$OWNER/$REPO/actions/workflows/$WORKFLOW_FILE_OR_ID/runs \
  --method GET -f head_sha=$COMMIT_SHA
```

**Comment by wn2try** [comment]
List recent workflow runs of the current repository, with the ID present:
```
gh run list -R ${{ github.repository }}
```
Filter options are available, here is the [manual](https://cli.github.com/manual/gh_run_list).

**Comment by sergeydobrodey** [comment]
See https://github.blog/changelog/2026-02-19-workflow-dispatch-api-now-returns-run-ids/
